### PR TITLE
Add Unit Tests for Player Widget and Related Services

### PR DESCRIPTION
**Pull request by Whitemoons**
## 📝 Description

This PR adds comprehensive unit and integration tests for multiple components of the application, primarily focusing on the player controls, data services, and caching functionality.

The goal of these tests is to improve code reliability, ensure proper widget behavior, validate service logic, and enhance overall test coverage. Additionally, new widget encapsulation improves modularity and reusability.

---

## 🔨 Changes Made

- Encapsulated slide thumbnail rendering
  - Improved handling of empty and error states
  - Introduced a new widget for handling slide thumbnail images
- Updated dependencies
  - Added `mockito` as a dev dependency for improved mocking and test isolation
- Added widget tests for player_widget
  - Verified rendering of player controls (play/pause, skip, back, captions, sync, timeline slider)
  - Tested icon display, callback triggers, and UI state transitions
  - Included integration tests for top control bar and central play control interactions
- Added unit tests for lecture_data_service
  - Validated fromJson parsing for metadata and transcript data
  - Tested error handling for invalid data
  - Added integration tests for full lecture data consistency
- Added unit tests for audio_service
  - Refactored service to allow dependency injection of AudioPlayer for testability
  - Added tests for playback controls, initialization, stream handling, and edge cases
- Added unit tests for pdf_cache_service
  - Verified cache operations, PDF document management, and retry limits
  - Tested edge conditions to ensure stable rendering behavior

---

## ✅ Checklist
- [x] Code compiles without errors
- [x] All tests passed locally
- [x] Added/updated tests
- [ ] Updated relevant documentation (README, Wiki, etc.)
- [x] Followed coding style and conventions

---

## 📷 Screenshots / Demos (if relevant)

---

## 🔗 Related Issues / PRs

Fixes #

---

## 👥 Reviewers

- @del2090 
- @soarhigh03 
---

## 📌 Notes for Reviewers

- Verify that test coverage has significantly improved across the tested modules.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -32,6 +32,9 @@ jobs:
     - name: Run lecture fetch unit tests
       run: flutter test test/lecture_fetch_test.dart
 
+    - name: Run player unit tests
+      run: flutter test test/features/player/core/pdf_cache_service_test.dart test/features/player/service/audio_service_test.dart test/features/player/models/lecture_data_service_test.dart test/features/player/player_widget_test.dart
```
Comment (del2090) [inline .github/workflows/ci.yml]: I believe you can just simply do `flutter test test/features/player`.

**Review by del2090 — CHANGES_REQUESTED** [review]
Everything looks fine, but please make the CI YAML file more readable by applying proposed changes.

**Review by del2090 — APPROVED** [review]
Thanks :)
